Quartz/Engine/Include/Quartz/Core/Internals/QuartzTypes.hpp

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Genten Studios
+// 
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the 
+// following conditions are met:
+// 
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the 
+// following disclaimer.
+// 
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the 
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+// 
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote 
+// products derived from this software without specific prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED 
+// WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A 
+// PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY 
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, 
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR 
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH 
+// DAMAGE.
+
+#pragma once
+
+#include <Quartz/Core/Internals/QuartzDefines.hpp>
+
+#if defined(QZ_MSVC)
+	using int8   = signed   __int8;
+	using uint8  = unsigned __int8;
+	using int16  = signed   __int16;
+	using uint16 = unsigned __int16;
+	using int32  = signed   __int32;
+	using uint32 = unsigned __int32;
+	using int64  = signed   __int64;
+	using uint64 = unsigned __int64;
+#else
+	using int8   = signed   char;
+	using uint8  = unsigned char;
+	using int16  = signed   short int;
+	using uint16 = unsigned short int;
+	using int32  = signed   int;
+	using uint32 = unsigned int;
+	using int64  =          long long int;
+	using uint64 = unsigned long long int;
+#endif
+
+using byte = int8;
```

Comment:
I would define just `byte` as `uint8` - I'd expect a byte to be the full 8 bits, and the concept of a unsigned byte doesn't really make sense as a byte is just a unit of data whereas unsigned is for numbers which need specific representations (for positive & negative etc..)